forage-android/src/main/java/com/joinforage/forage/android/ForageSDK.kt

```diff
@@ -32,20 +33,38 @@ object ForageSDK : ForageSDKApi {
     ): ForageApiResponse<String> {
         val currentEntry = panEntry
 
+        logger.i(
+            "Tokenizing Payment Method",
+            attributes = mapOf(
+                "merchant" to merchantAccount,
+                "customer_id" to customerId
+            )
+        )
+
         return when {
             shouldTokenize(currentEntry) -> TokenizeCardService(
                 okHttpClient = OkHttpClientBuilder.provideOkHttpClient(
                     bearerToken,
                     merchantAccount,
                     idempotencyKey = UUID.randomUUID().toString()
                 ),
-                httpUrl = ForageConstants.provideHttpUrl()
+                httpUrl = ForageConstants.provideHttpUrl(),
+                logger = logger
             ).tokenizeCard(
                 cardNumber = currentEntry.getPanNumber(),
                 customerId = customerId,
                 reusable = reusable
             )
-            else -> ForageApiResponse.Failure(listOf(ForageError(400, "invalid_input_data", "Invalid PAN entry")))
+            else -> {
+                logger.e(
+                    "PAN entry was invalid",
+                    attributes = mapOf(
+                        "merchant" to merchantAccount,
+                        "customer_id" to customerId
+                    )
+                )
+                ForageApiResponse.Failure(listOf(ForageError(400, "invalid_input_data", "Invalid PAN entry")))
```

Comment:
Soooo it looks like this code path is something we should think more about in the future. The `shouldTokenize` method checks if the PAN is valid or if we are in DEBUG mode. I think we may want to change this functionality to better fit how our merchants will use the field. In production, we want this check to exist so that we don't let people add a ton of garbage cards. In sandbox, we don't want this check to explicitly block cards from being added. However, we still want to report back to the developer through event handlers that the field is invalid. Maybe we should remove this check in general and just let the external dev manage this check themselves? I'm going to add a ticket for this to the Android GoPuff requests.